src/types.md

```diff
@@ -0,0 +1,183 @@
+# Questions about your types
+
+## My 'class' needs references to lots of other things in order to do its job, and that conflicts with other things which also hold similar references. What do I do?
+
+It's common in C++ to have objects which need to mutate other objects to get
+their work done. Those other objects may be accessed by several different
+objects.
```

Comment:
I feel this paragraph could be clearer -- there's a lot of "objects" flying around. How about this:

"It's common in C++ to have a class that contain mutable references to other objects; the class mutates those objects to do its work. Often, in fact, there are several classes that all hold a mutable reference to the same object. Here is a diagram that illustrates this:"